cmake/Platform/Windows/Packaging/Tests/test_installer.py

```diff
@@ -0,0 +1,217 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Pytest test configuration file.
+
+Example command to run these tests:
+
+pytest cmake/Platform/Windows/Packaging/Tests \
+    --capture=no \
+    --log-file=C:/workspace/test_installer.log \
+    --installer-uri=file:///C:/path/to/o3de_installer.exe \
+    --install-root=C:/O3DE/0.0.0.0 \
+    --project-path=C:/workspace/TestProject
+
+Alternately, the installer-uri can be an s3 or web URL. For example:
+--installer-uri=s3://bucketname/o3de_installer.exe
+--installer-uri=https://o3de.binaries.org/o3de_installer.exe
+
+"""
+import pytest
+import json
+from pathlib import Path
+from subprocess import TimeoutExpired
+
+
+@pytest.fixture(scope="session")
+def test_installer_fixture(context):
+    """ Installer executable succeeds. """
+    assert context.installer_path.is_file(), f"Invalid installer path {context.installer_path}"
+
+    # when the installer is run with timeout of 30 minutes
+    result = context.run([str(context.installer_path) , f"InstallFolder={context.install_root}", "/quiet"], timeout=30*60)
+
+    # the installer succeeds
+    assert result.returncode == 0, f"Installer failed with exit code {result.returncode}"
+
+    # the install root is created
+    assert context.install_root.is_dir(), f"Invalid install root {context.install_root}"
+
+
+@pytest.mark.parametrize(
+    "filename", [
+        pytest.param('Editor.exe'),
+        pytest.param('AssetProcessor.exe'),
+        pytest.param('MaterialEditor.exe'),
+        pytest.param('o3de.exe'),
+    ]
+)
+def test_binaries_exist(test_installer_fixture, context, filename):
+    """ Installer key binaries exist. """
+    assert (context.engine_bin_path / filename).is_file(), f"{filename} not found in {context.engine_bin_path}"
+
+
+@pytest.fixture(scope="session")
+def test_o3de_registers_engine_fixture(test_installer_fixture, context):
+    """ Engine is registered when o3de.exe is run. """
+    try:
+        result = context.run([str(context.engine_bin_path / 'o3de.exe')], timeout=15)
+
+        # o3de.exe should not close with an error code 
+        assert result.returncode == 0, f"o3de.exe failed with exit code {result.returncode}"
+    except TimeoutExpired as e:
+        # we expect to close the app ourselves
+        pass
+
+    engine_json_path = context.install_root / 'engine.json'
+    with engine_json_path.open('r') as f:
+        engine_json_data = json.load(f)
+
+    engine_name = engine_json_data['engine_name']
+
+    manifest_path = context.home_path / '.o3de' / 'o3de_manifest.json'
```

Comment:
FYI python pathlib takes care of slash normalization when there are embedded slashes as part of the string.
```
Python 3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 17:08:21) [MSC v.1927 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import pathlib
>>> myPath = pathlib.Path("Foo")
>>> myPath /= "Bar/Baz"
>>> print(myPath)
Foo\Bar\Baz
>>>
```